src/components/FlawForm.vue

```diff
@@ -506,12 +513,17 @@ const formDisabled = ref(false);
       />
       <div v-if="mode === 'edit'" class="border-top osim-flaw-form-section">
         <FlawComments
-          :comments="flaw.comments"
-          :taskKey="flaw.task_key"
-          :error="errors.comments"
-          :isSaving="isSaving"
-          @comment:addPublicComment="addPublicComment"
-          @disableForm="(value) => formDisabled = value"
+          :publicComments
+          :privateComments
+          :internalComments
+          :internalCommentsAvailable
+          :isLoadingInternalComments
+          :systemComments
+          :taskKey="flaw.task_key || ''"
+          :bugzillaLink
+          :isSaving
```

Comment:
What do you guys think about using a `v-bind` here to reduce the number of props? ([docs](https://vuejs.org/api/built-in-directives.html#v-bind))
Instead of returning all `xxxxxComments` props from the `useFlawModel`, return an object like `flawComments` that wraps all and pass that to the component.
With `v-bind` the object automatically expand to the component props
```js
// Inside useFlawModel
const flawComments = {
  publicComments,
  privateComments,
  internalComments,
  systemComments,
  internalCommentsAvailable,
  isLoadingInternalComments,
}

return { 
flawComments, 
addFlawComment,  
loadInternalComments,
// rest...
}
```
```js
// inside FlawForm
const {
flawComments,
addFlawComment,
loadInternalComments
// rest...
} = useFlawModel(props.flaw, onSaveSuccess);
```

```vue
<FlawComments v-bind="flawComments"
  :taskKey="flaw.task_key || ''"
  :bugzillaLink
  :isSaving
  @comment:addFlawComment="addFlawComment"
  @loadInternalComments="loadInternalComments"
/>
```

Function wise is the same but maybe a bit cleaner? I would like to know your opinion on this

Reply:
That's a good point. I think that approach is adding a bit of complexity, I'd weakly prefer to have some visibility on different props a component is receiving over wrapping them too much.

What I thought to do in order to reduce the props in this case is to use a single `flawComments` array on `useFlawCommentsModel` which contains all the comment types, and compute the individual comments arrays by type within the `FlawComments` template.

The reason I didn't do that is just in case want to add some new logic to a specific comment type only, in which case I think it would be proper to have it in the model.